# Factores cualitativos

**Nombre**: Valeria Estefanía Milke Loera

**Fecha**: 09-02-2026

**Expediente**: 739228
______________

In [3]:
import pandas as pd 
import numpy as np 
from sklearn.linear_model import LinearRegression
from scipy import stats
import statsmodels.api as sm

**Carga de datos**

In [5]:
df0 = pd.read_excel(r"C:\Users\valer\aguasfeo.xlsx")

In [6]:
df0.head()

,Tamaño,Hielo,Sabor,Score
0,grande,sin,kahlúa,8.2
1,grande,con,kahlúa,9.0
2,chica,sin,kahlúa,7.0
3,chica,con,kahlúa,7.5
4,grande,con,horchata,10.0


______________

**Modelo con codificación 0-1**

In [15]:
dff = df0.copy()
df = pd.DataFrame({
    "Tamaño":      (dff["Tamaño"] == "grande").astype(int),
    "Hielo":    (dff["Hielo"] == "con").astype(int),
    "Kahlúa":   (dff["Sabor"] == "kahlúa").astype(int),
    "Horchata": (dff["Sabor"] == "horchata").astype(int)
})

df.head(10)

,Tamaño,Hielo,Kahlúa,Horchata
0,1,0,1,0
1,1,1,1,0
2,0,0,1,0
3,0,1,1,0
4,1,1,0,1
5,0,0,0,1
6,0,1,0,1
7,1,0,0,1


In [8]:
x1 = df[["Tamaño","Hielo","Kahlúa","Horchata"]]
x = sm.add_constant(x1)
y = df0[["Score"]].values.reshape(-1,1)

In [9]:
xd = sm.add_constant(x)
model = sm.OLS(y,xd)
results = model.fit()
results.summary()

C:\Users\valer\anaconda3\Lib\site-packages\scipy\stats\_axis_nan_policy.py:430: UserWarning: `kurtosistest` p-value may be inaccurate with fewer than 20 observations; only n=8 observations were given.
  return hypotest_fun_in(*args, **kwds)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.837
Model:                            OLS   Adj. R-squared:                  0.714
Method:                 Least Squares   F-statistic:                     6.824
Date:                Wed, 11 Feb 2026   Prob (F-statistic):             0.0473
Time:                        21:10:30   Log-Likelihood:                -6.6425
No. Observations:                   8   AIC:                             21.29
Df Residuals:                       4   BIC:                             21.60
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          4.3000      0.320     13.417      0.000       3.410       5.190
Tamaño         2.4500      0.555      4.414      0.012       0.909       3.991
Hielo          0.5500      0.555      0.991      0.378      -0.991       2.091
Kahlúa         2.1250      0.320      6.631      0.003       1.235       3.015
Horchata       2.1750      0.320      6.787      0.002       1.285       3.065
==============================================================================
Omnibus:                        7.172   Durbin-Watson:                   1.431
Prob(Omnibus):                  0.028   Jarque-Bera (JB):                1.242
Skew:                          -0.048   Prob(JB):                        0.537
Kurtosis:                       1.072   Cond. No.                     3.17e+16
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The smallest eigenvalue is 1.64e-32. This might indicate that there are
strong multicollinearity problems or that the design matrix is singular.
"""

______________________

**Modelo con codificación -1–1**

In [17]:
df2 = pd.DataFrame({
    "Tamaño": df["Tamaño"].replace({1: -1, 0: 1}),
    "Hielo": df["Hielo"].replace({0: -1}),
    "Kahlúa": (dff["Sabor"] == "kahlúa").astype(int),
    "Horchata": (dff["Sabor"] == "horchata").astype(int)
})

df2.head(10)

,Tamaño,Hielo,Kahlúa,Horchata
0,-1,-1,1,0
1,-1,1,1,0
2,1,-1,1,0
3,1,1,1,0
4,-1,1,0,1
5,1,-1,0,1
6,1,1,0,1
7,-1,-1,0,1


In [18]:
x2 = df[["Tamaño","Hielo","Kahlúa","Horchata"]]
xx = sm.add_constant(x2)
yy = df0[["Score"]].values.reshape(-1,1)

In [19]:
xd2 = sm.add_constant(xx)
model2 = sm.OLS(yy,xd2)
results2 = model2.fit()
results2.summary()

C:\Users\valer\anaconda3\Lib\site-packages\scipy\stats\_axis_nan_policy.py:430: UserWarning: `kurtosistest` p-value may be inaccurate with fewer than 20 observations; only n=8 observations were given.
  return hypotest_fun_in(*args, **kwds)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.837
Model:                            OLS   Adj. R-squared:                  0.714
Method:                 Least Squares   F-statistic:                     6.824
Date:                Wed, 11 Feb 2026   Prob (F-statistic):             0.0473
Time:                        21:24:54   Log-Likelihood:                -6.6425
No. Observations:                   8   AIC:                             21.29
Df Residuals:                       4   BIC:                             21.60
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          4.3000      0.320     13.417      0.000       3.410       5.190
Tamaño         2.4500      0.555      4.414      0.012       0.909       3.991
Hielo          0.5500      0.555      0.991      0.378      -0.991       2.091
Kahlúa         2.1250      0.320      6.631      0.003       1.235       3.015
Horchata       2.1750      0.320      6.787      0.002       1.285       3.065
==============================================================================
Omnibus:                        7.172   Durbin-Watson:                   1.431
Prob(Omnibus):                  0.028   Jarque-Bera (JB):                1.242
Skew:                          -0.048   Prob(JB):                        0.537
Kurtosis:                       1.072   Cond. No.                     3.17e+16
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The smallest eigenvalue is 1.64e-32. This might indicate that there are
strong multicollinearity problems or that the design matrix is singular.
"""

____________________________
**Interpretación**
El modelo de regresión presenta un buen ajuste, con un R2 = 0.837, lo que indica que aproximadamente el 83.7% de la variabilidad del score es explicada por las variables incluidas. Además, el modelo en conjunto es estadísticamente significativo al 5%, por lo que al menos una de las variables influye en la calificación. Si analizamos l

En el análisis individual de los coeficientes:

Tamaño tiene un efecto positivo y estadísticamente significativo (p = 0.012), lo que indica que influye en el score.

Kahlúa (p = 0.003) y Horchata (p = 0.002) también presentan efectos positivos y significativos, por lo que el sabor impacta de manera importante en la calificación.

En contraste, Hielo tiene un p-value de 0.378, mayor que 0.05, y su intervalo de confianza incluye el cero. Esto indica que no existe evidencia estadística suficiente para afirmar que la presencia de hielo tenga un efecto significativo sobre el score.

En términos prácticos, el tamaño y el sabor son factores determinantes en la calificación del producto, mientras que la variable hielo no muestra un impacto estadísticamente relevante.